### PR TITLE
Add test for empty required fields for 'addresses/new'

### PR DESCRIPTION
**Pull request by Meyclem**
## Description

This PR adds a e2e test where we manually disable required field in the address form through Taiko.
Then we try to submit the form and expect for it to display the `Please fill in the above field.` error coming from Profile.

For now, the test will only be ran on staging, since it needs the real Connect.Profile and not the mock.

Here's the output of the test:
```
 PASS  tests/e2e/fill-incorrect-address-input-values.test.ts (41.554 s)
  Profile happy path
    ✓ it should display errors coming from Profile when required is disabled in address form (36940 ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        42.018 s
```

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [ ] **Bug fix** (non-breaking change which fixes an issue).
- [x] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## Issues

[CU-hd4b9u](https://app.clickup.com/t/hd4b9u)

## Checklist

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Comment by frieda-ewlines** [comment]
Task linked: [CU-hd4b9u Add missing E2E tests](https://app.clickup.com/t/hd4b9u)

File: tests/e2e/fill-incorrect-address-input-values.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    process.env.CONNECT_PROFILE_URL || CONFIG_VARIABLES.connectProfileUrl
+  ).includes("staging");
+
+  const localizedStrings = {
+    newAddress: locales.en["/account/profile/addresses/new"],
+    errors: locales.en["errors"],
+  };
+
+  beforeAll(async () => {
+    if (isStagingEnv) {
```
Comment (Meyclem) [inline tests/e2e/fill-incorrect-address-input-values.test.ts:31]: The test is not ran at all when `isStagingEnv` is set to `false` [1/2]

File: tests/e2e/fill-incorrect-address-input-values.test.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          ).elements(),
+        ).toHaveLength(3);
+      } catch (error) {
+        await screenshot({
+          path: "./tests/e2e/screenshots/fill-incorrect-address-input-values.png",
+        });
+        throw error;
+      }
+    } else {
+      printStep("Test auto success because the env is not staging.");
```
Comment (Meyclem) [inline tests/e2e/fill-incorrect-address-input-values.test.ts:89]: So I print a message to indicate that it's an "auto" succeeded test. [2/2]

File: tests/e2e/fill-incorrect-address-input-values.test.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      await openBrowser({
+        args: [
+          "--window-size=1440,1000",
+          "--no-sandbox",
+          "--start-maximized",
+          "--disable-dev-shm",
+        ],
+        headless: false,
+        observe: true,
+        observeTime: 1000,
```
Comment (Oluane) [inline tests/e2e/fill-incorrect-address-input-values.test.ts:41]: Is it intended to do it in observe mode ?

Comment (Meyclem) [reply tests/e2e/fill-incorrect-address-input-values.test.ts:41]: I did as we do on every other e2e test.
I tried without it but it failed ont the `for await` part and it failed 1 time on 5 with an `observeTime` of 500ms, so I let 1000ms.

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.85</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.88</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.86</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.9</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.89</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.89</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.89</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

File: tests/e2e/fill-incorrect-address-input-values.test.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  beforeAll(async () => {
+    if (isStagingEnv) {
+      await openBrowser({
+        args: [
+          "--window-size=1440,1000",
+          "--no-sandbox",
+          "--start-maximized",
+          "--disable-dev-shm",
+        ],
+        headless: false,
```
Comment (Fuzznimp) [inline tests/e2e/fill-incorrect-address-input-values.test.ts:39]:
```suggestion
        ],
        headless: true,
```

File: tests/e2e/fill-incorrect-address-input-values.test.ts (hunk 3 above)
Comment (Oluane) [reply tests/e2e/fill-incorrect-address-input-values.test.ts:41]: Alright, sometimes we don't have any other choice 👍

File: tests/e2e/fill-incorrect-address-input-values.test.ts (hunk 4 above)
Comment (Meyclem) [reply tests/e2e/fill-incorrect-address-input-values.test.ts:39]: 😅

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.88</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.9</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.9</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>

**Review by Oluane — APPROVED** [review]

**Review by Fuzznimp — APPROVED** [review]

**Comment by frieda-ewlines** [comment]
<details><summary>Click here to see the Lighthouse manifest</summary><br><p><strong>/:</strong></p><ul><li>Performance: 0.92</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/:</strong></p><ul><li>Performance: 0.95</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/profile/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 0.93</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/:</strong></p><ul><li>Performance: 0.94</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/email/new/:</strong></p><ul><li>Performance: 0.89</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/logins/phone/new/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/:</strong></p><ul><li>Performance: 0.91</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/security/sudo/:</strong></p><ul><li>Performance: 0.9</li><li>Accessibility: 0.98</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr><p><strong>/account/locale/:</strong></p><ul><li>Performance: 0.84</li><li>Accessibility: 1</li><li>Best practice: 1</li><li>SEO: 1</li></ul><hr></details>
